src/main/java/evelyn/command/TaskList.java

```diff
@@ -0,0 +1,216 @@
+        } catch (IOException e) {
+            System.err.println("Error reading/writing to file: " + e);
+        }
+    }
+
+    /**
+     * Gets a Task at a particular index from the list.
+     * @param index Index of the task to get.
+     * @return Returns a task at the specified index.
+     */
+    public Task getTask(int index) {
+        return (Task) list.get(index);
+    }
+
+    /**
+     * Marks the Task at a specified index as complete.
+     * @param index Index of the task in the list to be marked.
+     */
+    public void markTask(int index) {
+        Task task = (Task) list.get(index);
+        System.out.println(Ui.horizontalLine);
+        task.mark();
+        System.out.println(Ui.horizontalLine);
+    }
+
+    /**
+     * Unmarks a completed Task at a specified index.
+     * @param index Index of the task to be unmarked.
+     */
+    public void unmarkTask(int index) {
+        Task task = (Task) list.get(index);
+        System.out.println(Ui.horizontalLine);
+        task.unmark();
+        System.out.println(Ui.horizontalLine);
+    }
+
+    /**
+     * Gets the size of the current list
+     * @return Returns the size of the current list of tasks.
+     */
+    public int listSize() {
+        return list.size();
+    }
+
+    /**
+     * Saves all the tasks into a .txt file.
+     */
+    public void saveTasks() {
+        try {
+            this.storage.saveData();
+            for (int i = 0; i < this.listSize(); i++) {
+                Task task = (Task) this.getTask(i);
+                storage.writeToFile(task);
+            }
+        } catch (IOException e) {
+            System.err.println("Error saving to file: " + e);
+        }
+    }
+
+    /**
+     * Finds all tasks that contain the specified keyword
+     * @param keyword
+     */
+    public void find(String keyword) {
+        boolean foundSomething = false;
+       System.out.println(Ui.horizontalLine);
+       System.out.println("The following are all tasks and their indexes that contain the keyword: " + keyword);
+       for (int i = 0; i < list.size(); i++) {
+           String taskString = list.get(i).toString();
+           if (taskString.contains(keyword)) {
+               foundSomething = true;
+               System.out.println(i + ". " + taskString);
+           }
+       }
+
+       if (foundSomething) {
+           System.out.println(Ui.horizontalLine);
+       } else {
+           System.out.println("It seems like there is no task that matches your specified keyword." +
+                   " Try checking your spelling");
+           System.out.println(Ui.horizontalLine);
+       }
+    }
+
+    /**
+     * Deciphers the String data from the saved file and inputs it as a Task
+     * @param data String representation of a task form the save data.
+     * @throws IOException Exception thrown if the file data input is erroneous.
+     */
+    private void fileDataToList(String data) throws IOException {
```

Comment:
```suggestion
    private void writeFileDataToList(String data) throws IOException {
```
I think this might make the purpose of the method more obvious and also in line with the code quality standards mentioned!